edb/graphql/types.py

```diff
@@ -1022,27 +1203,36 @@ def _define_types(self):
                     description=self._get_description(t),
                 )
             else:
+                def _type_resolver(
+                    obj: GraphQLObjectType,
+                    info: GraphQLResolveInfo,
+                    _t: GraphQLAbstractType,
+                ) -> GraphQLObjectType:
+                    return obj
                 gqltype = GraphQLInterfaceType(
                     name=gql_name,
                     fields=partial(self.get_fields, t_name),
-                    resolve_type=lambda obj, info: obj,
+                    resolve_type=_type_resolver,
                     description=self._get_description(t),
                 )
-            self._gql_interfaces[t_name] = gqltype
+            # XXX: graphql-core typing annotations don't
+            #      seem to indicate that it is fine to mix
+            #      GraphQLObjectType with GraphQLInterfaceType
+            self._gql_interfaces[t_name] = gqltype  # type: ignore
```

Comment:
Conceptually interfaces are type-like, they represent potentially more than one type, but otherwise are used in most of the same places. They would have slightly different properties (such as that types can implement interfaces, but obviously not other types, etc.). I wonder if potentially asserting that what we use as `interfaces` is actually a `GraphQLInterfaceType` specifically would solve the typing complaint. Perhaps there are other places as well where we would need a similar assert. I can do that after you merge this PR, though.